boa/integrations/jupyter/browser.py

```diff
@@ -153,8 +153,31 @@ class BrowserEnv(NetworkEnv):
 
     def __init__(self, address=None, **kwargs):
         super().__init__(self._rpc, **kwargs)
-        self.signer = BrowserSigner(address, self._rpc)
-        self.set_eoa(self.signer)
+        self._given_address = address
+
+    @property
+    def signer(self):
+        """
+        Create a BrowserSigner instance. This needs to be recreated for every
+        call as the wallet address may change.
+        """
+        return BrowserSigner(self._given_address, self._rpc)
```

Comment:
so every time you grab `.signer` there are side effects? that seems weird. would it make more sense to override the relevant functions from NetworkEnv so that the signer gets refreshed?

Reply:
yeah it looks a bit weird. The alternative duplicates a bit of code, maybe we can remove the `address` field from the browser signer given it's undefined when created... :thinking: 
 https://github.com/vyperlang/titanoboa/pull/318/commits/71e1202382d95704e0b136b1ad78cac5dbf14329

Reply:
It's clever, but yea a bit of code duplication. Plus get_sender is called multiple times. Maybe override execute_code and deploy_code like

```
    self.signer.update()
    return super().execute_code(*args, **kwargs)
```
and similarly for deploy_code